benchmarks/utils/ArborXBenchmark_PointClouds.hpp

```diff
@@ -49,160 +47,163 @@ inline PointCloudType to_point_cloud_enum(std::string const &str)
+      : _point_cloud_type(point_cloud_type)
+  {}
+
+  // Assume random generates in (-1, 1) interval
+  template <typename Random>
+  KOKKOS_FUNCTION auto operator()(int i, Random const &random) const
   {
-    for (unsigned int i = 2 * axis; i < n; i += num_faces)
-      for (int d = 0; d < DIM; ++d)
-        random_points(i)[d] = (d != axis ? random() : -half_edge);
+    switch (_point_cloud_type)
+    {
+    case PointCloudType::filled_box:
+      return filledBoxPoint(random);
+    case PointCloudType::hollow_box:
+      return hollowBoxPoint(i, random);
+    case PointCloudType::filled_sphere:
+      return filledSpherePoint(random);
+    case PointCloudType::hollow_sphere:
+      return hollowSpherePoint(random);
+    default:
+      Kokkos::abort("ArborX: implementation bug");
+    }
+  }
 
-    for (unsigned int i = 2 * axis + 1; i < n; i += num_faces)
-      for (int d = 0; d < DIM; ++d)
-        random_points(i)[d] = (d != axis ? random() : +half_edge);
+private:
+  template <typename Random>
+  KOKKOS_FUNCTION auto filledBoxPoint(Random const &random) const
+  {
+    Point p;
+    for (int d = 0; d < DIM; ++d)
+      p[d] = random();
+    return p;
   }
-}
 
-template <class Point, typename... ViewProperties>
-void filledSphereCloud(double const radius,
-                       Kokkos::View<Point *, ViewProperties...> random_points)
-{
-  namespace KokkosExt = ArborX::Details::KokkosExt;
-  static_assert(
-      KokkosExt::is_accessible_from_host<decltype(random_points)>::value,
-      "The View should be accessible on the Host");
-  std::default_random_engine generator;
+  template <typename Random>
+  KOKKOS_FUNCTION auto hollowBoxPoint(int i, Random const &random) const
+  {
+    Point p;
+    // For 3D, the order is
+    // Indices: 0  1  2  3  4  5  6  7  8  9
+    // Axes   : 0- 0+ 1- 1+ 2- 2+ 0- 0+ 1- 1+
+    int axis = (i / 2) % DIM;
+    for (int d = 0; d < DIM; ++d)
+    {
+      if (d != axis)
+        p[d] = random();
+      else if (i % 2 == 0)
+        p[d] = -1;
+      else
+        p[d] = 1;
+    }
 
-  std::uniform_real_distribution<double> distribution(-radius, radius);
-  auto random = [&distribution, &generator]() {
-    return distribution(generator);
-  };
+    return p;
+  }
 
-  unsigned int const n = random_points.extent(0);
-  constexpr auto DIM = ArborX::GeometryTraits::dimension<Point>::value;
-  for (unsigned int i = 0; i < n; ++i)
+  template <typename Random>
+  KOKKOS_FUNCTION auto filledSpherePoint(Random const &random) const
   {
-    bool point_accepted = false;
-    while (!point_accepted)
+    Point p;
+    auto const radius = 1;
+    Coordinate norm;
+    do
     {
-      Point p;
-      double norm = 0.f;
+      norm = 0;
       for (int d = 0; d < DIM; ++d)
       {
-        double v = random();
-        p[d] = v;
-        norm += v * v;
+        p[d] = random();
+        norm += p[d] * p[d];
       }
-      norm = std::sqrt(norm);
+      norm = Kokkos::sqrt(norm);
```

Comment:
You could only take the square root at the end once and compare to the squared radius instead.

Reply:
Done.